crates/ruff_python_parser/src/string.rs

```diff
@@ -304,40 +305,27 @@ impl<'a> StringParser<'a> {
                     }
 
                     let ret = if self_documenting {
-                        // TODO: range is wrong but `self_documenting` needs revisiting beyond
-                        // ranges: https://github.com/astral-sh/ruff/issues/5970
-                        vec![
-                            Expr::from(ast::ExprConstant {
-                                value: Constant::Str(expression.clone() + "="),
-                                kind: None,
-                                range: self.range(start_location),
-                            }),
-                            Expr::from(ast::ExprConstant {
-                                value: trailing_seq.into(),
-                                kind: None,
-                                range: self.range(start_location),
-                            }),
-                            Expr::from(ast::ExprFormattedValue {
-                                value: Box::new(
-                                    parse_fstring_expr(&expression, start_location).map_err(
-                                        |e| {
-                                            FStringError::new(
-                                                InvalidExpression(Box::new(e.error)),
-                                                start_location,
-                                            )
-                                        },
-                                    )?,
-                                ),
-                                conversion: if conversion == ConversionFlag::None && spec.is_none()
-                                {
-                                    ConversionFlag::Repr
-                                } else {
-                                    conversion
-                                },
-                                format_spec: spec,
-                                range: self.range(start_location),
+                        let value =
+                            parse_fstring_expr(&expression, start_location).map_err(|e| {
+                                FStringError::new(
+                                    InvalidExpression(Box::new(e.error)),
+                                    start_location,
+                                )
+                            })?;
+                        let leading =
```

Comment:
this feels a little hacky but was my best idea. suggestions welcome

Reply:
Which part do you find hacky?

Reply:
pulling slices out of `expression`

Reply:
I think it's fine. I ultimately want to re-write this entire file and use `Cursor` and use string slices from the source text directly everywhere. But @dhruvmanila may make this whole code redundant before when implementing the 3.12 F-string changes.